### PR TITLE
add user expected etcd image

### PR DESCRIPTION
**Pull request by wilhelmguo**
for user Get image from private registry

### CODE REVIEW

**Comment by hongchaodeng** [comment]
What's your use case? Why do you want custom image?

**Comment by futureskywei** [comment]
@hongchaodeng Sometimes docker can not access to the quay.io, For example, the firewall 。

File: pkg/spec/cluster.go [hunk 1]
```diff
@@ -70,6 +71,10 @@ type ClusterSpec struct {
 	// The vaild range of the size is from 1 to 7.
 	Size int `json:"size"`
 
+	// Image is the expected etcd image name
+	// for user Get image from private registry
+	// default is quay.io/coreos/etcd
+	Image string `json:"image"`
```
Comment (xiang90) [inline pkg/spec/cluster.go]: BaseImage? We might attach version tags, for example quay.io/coreos/etcd:v3.0.17

**Comment by hongchaodeng** [comment]
Closing this due to inactivity and favor https://github.com/coreos/etcd-operator/pull/1162

**Pull request closed without merging**